### ISSUE TITLE
Function level parallelization

### ISSUE THREAD

**Issue by SweetVishnya** [craigahobbs/unittest-parallel#8]
Is it possible to detect all test functions, split them in chuncks and run each chunk in a separate processes? For Linux fork can be used.

**Comment by SweetVishnya** [comment]
Hmmm, after [this](https://github.com/craigahobbs/unittest-parallel/issues/7#issuecomment-844392909) fix. All unittest functions run in parallel even when they are in same class.

**Comment by SweetVishnya** [comment]
Seems that unittest-parallel 1.2.0 solves my task))) So, I can just:
```
python -m pip install unittest-parallel==1.2.0
```

**Comment by craigahobbs** [comment]
Correct, that will install unittest-parallel globally (alongside python3). If you don't want to install globally, you can install in a venv:

``` sh
python3 -m venv env
env/bin/pip install unittest-parallel
. env/bin/activate
unitest-parallel -v -t src -s src/tests
deactivate
```

Alternatively, you can skip activate/deactivate:

```
env/bin/unitest-parallel -v -t src -s src/tests
```

**Comment by SweetVishnya** [comment]
> Hmmm, after [this](https://github.com/craigahobbs/unittest-parallel/issues/7#issuecomment-844392909) fix all unittest functions run in parallel even when they are in same class.

My main point was that now your tool runs all functions in one TestCase in parallel.

**Comment by craigahobbs** [comment]
That is surprising, and you are correct. That's just what Python's unittest.TestLoader.discover method is returning - one TestCase per test method. I've tested this with setUp and tearDown methods, and it all works, so this is a good thing. :)

**Comment by craigahobbs** [comment]
This change introduced an issue with setUpClass/tearDownClass:

https://github.com/craigahobbs/unittest-parallel/issues/9
